pandas/core/strings.py

```diff
@@ -1910,23 +1936,34 @@ def _make_str_accessor(self):
             raise AttributeError("Can only use .str accessor with string "
                                  "values, which use np.object_ dtype in "
                                  "pandas")
-        elif isinstance(self, Index):
+        elif isinstance(data, Index):
             # can't use ABCIndex to exclude non-str
 
             # see scc/inferrence.pyx which can contain string values
             allowed_types = ('string', 'unicode', 'mixed', 'mixed-integer')
-            if self.inferred_type not in allowed_types:
+            if data.inferred_type not in allowed_types:
                 message = ("Can only use .str accessor with string values "
                            "(i.e. inferred_type is 'string', 'unicode' or "
                            "'mixed')")
                 raise AttributeError(message)
-            if self.nlevels > 1:
+            if data.nlevels > 1:
                 message = ("Can only use .str accessor with Index, not "
                            "MultiIndex")
                 raise AttributeError(message)
-        return StringMethods(self)
+        return StringAccessor(data)
+
+
+StringAccessor = StringMethods  # Alias to mirror CategoricalAccessor
+
+
+# TODO: This is only mixed in to Index (this PR takes it out of Series)
+# and the _dir_additions/_dir_deletions won't play nicely with
+# any other class this gets mixed into that *does* implement its own
+# _dir_additions/_dir_deletions.  This should be deprecated.
+class StringAccessorMixin(object):
```

Comment:
Great; wasn't sure if it should be saved for a separate PR.  Doing so requires copy/pasting `_accessors`, `_dir_deletions`, and `_dir_additions` from `Series` (which I see as a good thing, per #8162 , #17061)